source/client-side-encryption/client-side-encryption.md

```diff
@@ -0,0 +1,2475 @@
+
+## Integrating with libmongocrypt
+
+Each ClientEncryption instance MUST have one `libmongocrypt_handle`.
+
+[The libmongocrypt C API documentation](https://github.com/mongodb/libmongocrypt/blob/master/src/mongocrypt.h) For
+information on how to initialize, encrypt, decrypt with libmongocrypt.
+
+[The Guide to Integrating libmongocrypt](https://github.com/mongodb/libmongocrypt/blob/master/integrating.md) For
+information about integrating the libmongocrypt library in a driver.
+
+libmongocrypt exposes logging capabilities. If a driver provides a logging mechanism, it MUST enable this logging and
+integrate. E.g. if your driver exposes a logging callback that a user can set, it SHOULD be possible to get log messages
+from libmongocrypt.
+
+Drivers MUST propagate errors from libmongocrypt in whatever way is idiomatic to the driver (exception, error object,
+etc.). These errors MUST be distinguished in some way (e.g. exception type) to make it easier for users to distinguish
+when a command fails due to client side encryption.
+
+> \[!INDEX\]
+>
+> ## Enabling Command Marking with the [crypt_shared](#crypt_shared) Library
+>
+> The MongoDB Enterprise distribution includes a dynamic library named `mongo_crypt_v1` (with the appropriate file
+> extension or filename suffix for the host platform). This library will be loaded by [libmongocrypt](#libmongocrypt)
+> when the `mongocrypt_init` function is invoked
+> [(from the libmongocrypt C API)](https://github.com/mongodb/libmongocrypt/blob/master/src/mongocrypt.h) based on the
+> search criteria that are provided by the driver.
+>
+> [libmongocrypt](#libmongocrypt) allows the driver to specify an arbitrary list of directory \[search paths\](#search
+> paths) in which to search for the [crypt_shared](#crypt_shared) dynamic library. The user-facing API does not expose
+> this full search path functionality. This extended search path customization is intended to facilitate driver testing
+> with [crypt_shared](#crypt_shared) (Refer: [Search Paths for Testing](#search-paths-for-testing) and
+> `Path Resolution Behavior`).
+>
+> \[!NOTE\]
+>
+> The driver MUST NOT manipulate or do any validation on the [crypt_shared](#crypt_shared) path options provided in
+> [extraOptions](#extraoptions). They should be passed through to [libmongocrypt](#libmongocrypt) unchanged.
+
+<div id="search path">
+
+### Setting Search Paths
+
+For the user-facing API the driver MUST append the literal string - Ts:`"$SYSTEM"` to the search paths for the
+`libmongocrypt_handle` if `bypassAutoEncryption` is not set to `true`, and MUST NOT append to the search path if it is
+set to `true` or if the [libmongocrypt](#libmongocrypt) instance is used for explicit encryption only (i.e. on the
+ClientEncryption class). For purposes of testing, a driver may use a different set of search paths.
+
+**Explanation**
+
+The \[search paths\](#search paths) array in [libmongocrypt](#libmongocrypt) allows the driver to customize the way that
```

Comment:
Unnecessary escaping and invalid anchor (includes a space).

Reply:
Fixed